static_frame/core/quilt.py

```diff
@@ -466,29 +476,67 @@ def rename(self, name: NameType) -> 'Quilt':
 
     #---------------------------------------------------------------------------
 
-    def __repr__(self) -> str:
-        '''Provide a display of the :obj:`Quilt` that does not realize the entire :obj:`Frame`.
-        '''
-        if self.name:
-            header = f'{self.__class__.__name__}: {self.name}'
-        else:
-            header = self.__class__.__name__
-        return f'<{header} at {hex(id(self))}>'
-
+    @doc_inject()
     def display(self,
             config: tp.Optional[DisplayConfig] = None,
             *,
             style_config: tp.Optional[StyleConfig] = None,
             ) -> Display:
-        '''Provide a :obj:`Frame`-style display of the :obj:`Quilt`.
+        '''{doc}
+
+        Args:
+            {config}
         '''
         if self._assign_axis:
             self._update_axis_labels()
-        return self.to_frame().display( #type: ignore
-                config,
+
+        bus_columns = False # Flag to strip off column dtype info
+
+        if self._axis == 0:
+            if not self._retain_labels:
+                index = self.index.rename("Concatenated")
+            else:
+                index = self._bus.index.rename("Frames")
+            columns = self.columns.rename("Aligned")
+        else:
+            index = self.index.rename("Aligned")
+            if not self._retain_labels:
+                columns = self.columns.rename("Concatenated")
+            else:
+                columns = self._bus.index.rename("Frames")
+                bus_columns = True
+
+        config = config or DisplayConfig()
+
+        def placeholder_gen() -> tp.Iterator[tp.Iterable[tp.Any]]:
+            assert config is not None
+            yield from repeat(tuple(repeat(config.cell_placeholder, times=len(index))), times=len(columns))
+
+        d = Display.from_params(
+                index=index,
+                columns=columns,
+                header=DisplayHeader(self.__class__, self.name),
+                column_forward_iter=placeholder_gen,
+                column_reverse_iter=placeholder_gen,
+                column_default_iter=placeholder_gen,
+                config=config,
                 style_config=style_config,
                 )
 
+        # Strip out the dtype information!
+        if config.type_show:
+            if bus_columns:
+                # First Column Row -> last element is the dtype of the column
+                # Guaranteed to not be index hierarchy as buses cannot have index hierarchies
+                d._rows[1].pop()
+
+            # Since placeholder_gen is not a ndarray, there is no dtype to append in the final row
+            # However, in the case of a center ellipsis being added, an ellipsis will be
+            # awkwardly placed direclty adjacent to the index dtype information.
+            if d._rows[-1][-1] is Display.CELL_ELLIPSIS:
+                d._rows[-1].pop()
```

Comment:
I wish there was a better way, but I understand there may not be at this time.

I think we want to use string equality here rather than object id.

```suggestion
            if d._rows[-1][-1] == Display.CELL_ELLIPSIS:
                d._rows[-1].pop()
```